portality/tasks/old_data_cleanup.py

```diff
@@ -0,0 +1,92 @@
+import datetime
+from typing import Type
+
+from portality.background import BackgroundTask
+from portality.core import app
+from portality.dao import DomainObject
+from portality.decorators import write_required
+from portality.lib.es_queries import ES_DATETIME_FMT
+from portality.models import Notification, BackgroundJob
+from portality.tasks.helpers import background_helper
+from portality.tasks.redis_huey import schedule, long_running
+
+target_queue = long_running
+
+
+def _clean_old_data(domain_class: Type[DomainObject], retention_days,
+                    datetime_field='created_date',
+                    logger_fn=None, ):
+    if logger_fn is None:
+        logger_fn = print
+
+    last_retention_date = datetime.datetime.now() - datetime.timedelta(days=retention_days)
+    query = {
+        'query': {
+            'range': {
+                datetime_field: {
+                    'lte': last_retention_date.strftime(ES_DATETIME_FMT),
+                }
+            }
+        }
+    }
```

Comment:
I'd like to stick to our pattern of encapsulating queries in their own objects (kind of like a Prepared Statement in SQL), rather than defining them inline, which helps with upgrades to ES, and general code organisation.  Could we move this out to an object like

```
class RetentionQuery:
    def __init__(self, last_retention_date):
        self._lrd = last_rentention_date

    def query(self):
        # returns the query dict
        return { "query": ... }
```

So then you would use that like

```
q = RetentionQuery(last_retention_date)
num_record = domain_class.hit_count(q.query())
```